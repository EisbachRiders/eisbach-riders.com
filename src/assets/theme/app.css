html {
  height: 100%;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Raleway', sans-serif;
  background-color: #ffffff;
  overflow-x: hidden;
}

a {
  text-decoration: none;
}
