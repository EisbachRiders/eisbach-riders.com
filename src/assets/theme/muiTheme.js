import { createMuiTheme } from '@material-ui/core/styles'
import './App.css'
require('typeface-open-sans')
require('typeface-contrail-one')

const status = {
  white: '#ffffff',
  offWhite: '#F7F5F5',
  black: '#0b0c10',
  blue: '#1f2833',
  grey: '#c5c6c7',
  greyDk: '#404040',
  teal: '#66fcf1',
  tealDk: '#45a29e',
  green: '#66fca6',
  yellow: '#fcf166',
  red: '#e25b65',
}

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: status.teal,
      main: status.teal,
      dark: status.tealDk,
    },
    secondary: {
      light: status.blue,
      main: status.blue,
      dark: status.blue,
    },
    error: { main: status.red },
  },
  status,
  typography: {
    fontFamily: "'Open Sans', sans-serif",
    fontSize: 14,
  },
  overrides: {
    MuiTypography: {
      colorPrimary: {
        color: status.greyDk,
      },
      colorSecondary: {
        color: status.grey,
      },
      colorTextSecondary: {
        color: status.white,
      },
    },
  },
})

export default theme
