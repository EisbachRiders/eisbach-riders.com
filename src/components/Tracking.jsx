import React, { useState, Fragment } from 'react'
import { Link } from 'gatsby'
import { useTranslation } from 'react-i18next'
import { makeStyles } from '@material-ui/styles'
import Button from '@material-ui/core/Button'
import Snackbar from '@material-ui/core/Snackbar'
import SnackbarContent from '@material-ui/core/SnackbarContent'

const useStyles = makeStyles(theme => ({
  snackbar: {
    left: 0,
    right: 0,
    transform: 'none',
  },
  snackbarContent: {
    borderRadius: 0,
    width: '100%',
    minWidth: '0',
    maxWidth: 'none',
    paddingLeft: 30,
    paddingRight: 30,
    backgroundColor: theme.palette.common.black,
    [theme.breakpoints.up('md')]: {
      paddingLeft: 60,
      paddingRight: 60,
    },
  },
  message: {
    flexBasis: '100%',
    [theme.breakpoints.up('sm')]: {
      flexBasis: '60%',
    },
    [theme.breakpoints.up('lg')]: {
      flexBasis: '80%',
    },
  },
  action: {
    marginLeft: 'none',
    marginRight: 'none',
    paddingLeft: 0,
  },
  button: {
    marginLeft: 35,
    paddingLeft: 50,
    paddingRight: 50,
  },
  buttonDecline: {
    color: theme.palette.common.white,
  },
  link: {
    color: theme.palette.common.white,
    paddingLeft: 8,
    textTransform: 'capitalize',
    textDecoration: 'underline',
    [theme.breakpoints.up('sm')]: {
      paddingLeft: 0,
    },
    [theme.breakpoints.up('md')]: {
      paddingLeft: 8,
    },
  },
}))

function Tracking() {
  const classes = useStyles()
  const { t } = useTranslation()
  const [isSnackbarOpen, setSnackbar] = useState(true)

  const handleSnackbarAccept = () => {
    setSnackbar(false)
  }

  const handleSnackbarDecline = () => {
    window['ga-disable-UA-130658859-1'] = true
    setSnackbar(false)
  }

  return (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      open={isSnackbarOpen}
      onClose={handleSnackbarAccept}
      className={classes.snackbar}
    >
      <SnackbarContent
        className={classes.snackbarContent}
        classes={{ message: classes.message, action: classes.action }}
        aria-describedby="client-snackbar"
        onClose={handleSnackbarAccept}
        message={
          <span id="client-snackbar">
            {t('tracking.message')}
            <Link to="/privacy/" className={classes.link}>
              {t('tracking.privacyLink')}
            </Link>
          </span>
        }
        action={[
          <Fragment key={`snackbar`}>
            <Button
              className={classes.buttonDecline} //eslint-disable-next-line
              href="javascript:gaOptout();"
              onClick={handleSnackbarDecline}
            >
              {t('tracking.decline')}
            </Button>
            <Button
              color="secondary"
              variant="contained"
              className={classes.button}
              onClick={handleSnackbarAccept}
            >
              {t('tracking.accept')}
            </Button>
          </Fragment>,
        ]}
      />
    </Snackbar>
  )
}

export default Tracking
