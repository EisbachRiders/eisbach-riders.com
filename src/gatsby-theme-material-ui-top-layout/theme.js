import { createMuiTheme } from "@material-ui/core/styles"
import "../app.css"
import "../../static/fonts/fonts.css"

// fin: "#00d7a2",
// finDk: "#00be8f",
// finLt: "#e2f9ef",
// greyLt2: "#eeeff0",
// greyLt: "#d5d9db",
// grey: "#818C93",
// greyBlue: "#f3f8fc",
// greenLt1: "#6dff97",
// greenLt2: "#00cf3c",
// cream: "#f7f5ee",

const color = {
  // for shop colors
  blue: "#004AB5",
  green: "#00942b",
  yellow: "#fce566",
  orange: "#ff8000",
  pink: "#ff0080",
  red: "#FF0012",
  redLt: "#ff4e5b",
  // theme colors
  white: "#ffffff",
  black: "#000000",
  almostBlack: "#363738", // typography
  textLight: "#838588",
  gray: "#ebedee",
  primaryBlue: "#0062f0",
  primaryBlueLt: "#3f8eff",
  gold: "#E6C55E",
  goldLt: "#f0dda0",
}

const theme = createMuiTheme({
  palette: {
    type: "light",
    primary: {
      light: color.goldLt,
      main: color.gold,
      dark: color.goldLt,
    },
    secondary: {
      light: color.primaryBlueLt,
      main: color.primaryBlue,
      dark: color.primaryBlueLt,
    },
    error: { main: color.red },
  },
  color,
  typography: {
    fontFamily: "'Raleway', sans-serif",
    h1: {
      letterSpacing: 2,
    },
    h2: {
      letterSpacing: 2,
    },
    h3: {
      letterSpacing: 2,
    },
    h4: {
      letterSpacing: 2,
    },
    h5: {
      letterSpacing: 2,
    },
    body1: {
      letterSpacing: 1.5,
      fontSize: 14,
    },
    body2: {
      letterSpacing: 1.5,
      fontSize: 12,
    },
    button: {
      fontFamily: "'Raleway', sans-serif",
      letterSpacing: 3,
    },
  },
  overrides: {
    MuiButton: {
      root: {
        padding: "9px 27px",
        borderRadius: 0,
      },
      contained: {
        boxShadow: "none",
      },
    },
  },
})

export default theme
