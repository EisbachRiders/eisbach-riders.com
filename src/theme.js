import { createTheme } from "@material-ui/core/styles"
import "./css/app.css"

// fin: "#00d7a2",
// finDk: "#00be8f",
// finLt: "#e2f9ef",
// grayLt2: "#eeeff0",
// grayLt: "#d5d9db",
// gray: "#818C93",
// grayBlue: "#f3f8fc",
// greenLt1: "#6dff97",
// greenLt2: "#00cf3c",
// cream: "#f7f5ee",

const color = {
  // for shop colors
  blue: "#004AB5",
  green: "#00942b",
  yellow: "#fce566",
  orange: "#ff8000",
  pink: "#ff0080",
  red: "#FF0012",
  redLt: "#ff4e5b",
  seaGlass: "#5F5850",
  // theme colors
  white: "#ffffff",
  black: "#0B0C10",
  blackLt: "#1F2833",
  gray: "#C5C6C7",
  // teal: "#66FCF1",
  teal: "#00d7a2",
  tealDk: "#45A29E",
  test: "#F7F680",
  cream: "#F1F0EB",
  grayLt: "#f7f7f9",
  blueGray: "#cbd3cf",
  steel: "#505D72",
}

const theme = createTheme({
  palette: {
    type: "light",
    primary: {
      light: color.tealDk,
      main: color.teal,
      dark: color.tealDk,
    },
    secondary: {
      light: "#9d9fa0",
      main: color.gray,
      dark: "#9d9fa0",
    },
    error: { main: color.red },
  },
  color,
  typography: {
    fontFamily: ["Raleway", "sans-serif"].join(","),
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          padding: "9px 27px",
          borderRadius: 0,
        },
        contained: {
          boxShadow: "none",
        },
      },
    },
  },
})

export default theme
