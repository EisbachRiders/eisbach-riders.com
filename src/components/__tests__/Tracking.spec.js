import React from 'react'
import { render } from 'react-testing-library'
import Tracking from '../Tracking'
import { ThemeProvider } from '@material-ui/styles'
import theme from '../../assets/theme/muiTheme'

describe('Tracking', () => {
  it('renders correctly', () => {
    const mockProps = {
      t: () => '',
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <Tracking {...mockProps} />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
