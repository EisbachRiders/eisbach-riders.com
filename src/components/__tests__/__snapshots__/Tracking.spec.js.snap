// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracking renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomCenter-3 Hook-snackbar-1o19kbz"
      >
        <div
          aria-describedby="client-snackbar"
          class="MuiTypography-root-38 MuiTypography-body2-39 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8 Hook-snackbarContent-1xoobo1"
          role="alertdialog"
          style="-webkit-transform: translate(0, 0); transform: translate(0, 0); -webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiSnackbarContent-message-9 Hook-message-1dq2bje"
          >
            <span
              id="client-snackbar"
            >
              tracking.message
              <a
                class="Hook-link-15a5j2r"
                href="/privacy/"
              >
                tracking.privacyLink
              </a>
            </span>
          </div>
          <div
            class="MuiSnackbarContent-action-10 Hook-action-17qmw59"
          >
            <a
              class="MuiButtonBase-root-85 MuiButton-root-68 MuiButton-text-70 Hook-buttonDecline-xcsokg"
              href="javascript:gaOptout();"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label-69"
              >
                tracking.decline
              </span>
              <span
                class="MuiTouchRipple-root-88"
              />
            </a>
            <button
              class="MuiButtonBase-root-85 MuiButton-root-68 MuiButton-contained-76 MuiButton-containedSecondary-78 Hook-button-12rvs9h"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-69"
              >
                tracking.accept
              </span>
              <span
                class="MuiTouchRipple-root-88"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root-1 MuiSnackbar-anchorOriginBottomCenter-3 Hook-snackbar-1o19kbz"
    >
      <div
        aria-describedby="client-snackbar"
        class="MuiTypography-root-38 MuiTypography-body2-39 MuiPaper-root-11 MuiPaper-elevation6-19 MuiSnackbarContent-root-8 Hook-snackbarContent-1xoobo1"
        role="alertdialog"
        style="-webkit-transform: translate(0, 0); transform: translate(0, 0); -webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message-9 Hook-message-1dq2bje"
        >
          <span
            id="client-snackbar"
          >
            tracking.message
            <a
              class="Hook-link-15a5j2r"
              href="/privacy/"
            >
              tracking.privacyLink
            </a>
          </span>
        </div>
        <div
          class="MuiSnackbarContent-action-10 Hook-action-17qmw59"
        >
          <a
            class="MuiButtonBase-root-85 MuiButton-root-68 MuiButton-text-70 Hook-buttonDecline-xcsokg"
            href="javascript:gaOptout();"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label-69"
            >
              tracking.decline
            </span>
            <span
              class="MuiTouchRipple-root-88"
            />
          </a>
          <button
            class="MuiButtonBase-root-85 MuiButton-root-68 MuiButton-contained-76 MuiButton-containedSecondary-78 Hook-button-12rvs9h"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-69"
            >
              tracking.accept
            </span>
            <span
              class="MuiTouchRipple-root-88"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
